Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -356,185 +472,304 @@ void LoadPSIMuonBin::assignOutputWorkspaceParticulars(
+  addToSampleLog("Number of Spectra", m_header.numberOfHistograms,
+                 outputWorkspace);
+  addToSampleLog("Mon number of events", m_header.monNumberOfevents,
+                 outputWorkspace);
+  addToSampleLog("Mon Period", m_header.periodOfMon, outputWorkspace);
 
   if (m_header.monLow[0] == 0 && m_header.monHigh[0] == 0) {
-    logAlg->setProperty("LogType", "String");
-    logAlg->setProperty("LogName", "Mon Low");
-    logAlg->setProperty("LogText", "0");
-    logAlg->executeAsChildAlg();
-
-    logAlg->setProperty("LogType", "String");
-    logAlg->setProperty("LogName", "Mon High");
-    logAlg->setProperty("LogText", "0");
-    logAlg->executeAsChildAlg();
+    addToSampleLog("Mon Low", "0", outputWorkspace);
+    addToSampleLog("Mon High", "0", outputWorkspace);
   } else {
     for (auto i = 0u; i < 4; ++i) {
       if (m_header.monLow[i] == 0 || m_header.monHigh[i] == 0)
         break;
-      logAlg->setProperty("LogType", "String");
-      logAlg->setProperty("LogName", "Mon Low" + std::to_string(i));
-      logAlg->setProperty("LogText", std::to_string(m_header.monLow[i]));
-      logAlg->executeAsChildAlg();
-
-      logAlg->setProperty("LogType", "String");
-      logAlg->setProperty("LogName", "Mon High + i");
-      logAlg->setProperty("LogText", std::to_string(m_header.monHigh[i]));
-      logAlg->executeAsChildAlg();
+      addToSampleLog("Mon Low " + std::to_string(i), m_header.monLow[i],
+                     outputWorkspace);
+      addToSampleLog("Mon High" + std::to_string(i), m_header.monHigh[i],
+                     outputWorkspace);
     }
   }
 
-  // Mon Deviation
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "Mon Deviation");
-  logAlg->setProperty("LogText", m_header.monDeviation);
-  logAlg->executeAsChildAlg();
+  addToSampleLog("Mon Deviation", m_header.monDeviation, outputWorkspace);
 
   if (m_header.realT0[0] != 0) {
-    for (const float &i : m_header.realT0) {
-      if (i == 0)
+    // 16 is the max size of realT0
+    for (auto i = 0u; i < 16; ++i) {
+      if (m_header.realT0[i] == 0)
         break;
-      logAlg->setProperty("LogType", "String");
-      logAlg->setProperty("LogName", "realT0 + i");
-      logAlg->setProperty("LogText", std::to_string(i));
+      addToSampleLog("realT0 " + std::to_string(i), m_header.realT0[i],
+                     outputWorkspace);
+    }
+  }
+
+  // Read in the temperature file if provided/found
+  try {
+    readInTemperatureFile(outputWorkspace);
+  } catch (std::invalid_argument &e) {
+    g_log.warning("Temperature file could not be loaded: " +
+                  std::string(e.what()));
+  } catch (std::runtime_error &e) {
+    g_log.warning("Temperature file could not be loaded:" +
+                  std::string(e.what()));
+  }
+}
+
+void LoadPSIMuonBin::processTitleHeaderLine(const std::string &line) {
+  bool titlesFound = false;
+  std::string foundTitles = "";
+  for (const auto &charecter : line) {
```

Comment:
instead of having a continue we could make this a function and just have a return. It does not need any mantid stuff so could go into an anonymous namespace.

Reply:
The continue actually serves no purpose so I have removed it but I would say that abstracting away a single-use piece of code to a function doesn't make it more readable to me.

I've left it in a new function in an anonymous namespace just above the current function.